src/porepy/models/energy_balance.py

```diff
@@ -657,8 +717,14 @@ def bc_values_fourier(self, subdomains: list[pp.Grid]) -> pp.ad.DenseArray:
             Ad array representing the boundary condition values for the Fourier flux.
 
         """
-        num_faces = sum([sd.num_faces for sd in subdomains])
-        return pp.wrap_as_ad_array(0, num_faces, "bc_values_fourier")
+        boundary_projection = pp.ad.BoundaryProjection(self.mdg, subdomains=subdomains)
+        boundary_grids = self.subdomains_to_boundary_grids(subdomains)
+        temperature_dirichlet = self.temperature(boundary_grids)
+        flux_neumann = self.fourier_flux(boundary_grids)
+        result = temperature_dirichlet + flux_neumann  # TODO heat conductivity ?
+        result = boundary_projection.boundary_to_subdomain @ result
```

Comment:
This violates the pp guideline about variable redefinition. Please improve variable naming.